### ISSUE TITLE
[BUG/ISSUE] GEOS-Chem restarts don't save correctly at unusual intervals (e.g. month and a day)

### ISSUE THREAD

**Issue by drewpendergrass** [geoschem/geos-chem#1020]
### What institution are you from?
Harvard

### Description of the problem
For a data assimilation application, I need to run GEOS-Chem for unusual short intervals, such as a month and a day. GEOS-Chem does indeed run for this interval in input.geos (e.g. December 1 2018 through January 2 2019) but the restart, rather than being saved for Jan 2, will save for Jan 1. (The HISTORY.rc setting is set to 'End'). Even when I change the HISTORY.rc frequency/duration to a month and a day, rather than end, this same issue occurs.

### Description of troubleshooting performed
I currently work around this problem by running GC for standard intervals (a month) and then execute the run script again for the shorter interval (a day).

### GEOS-Chem version
13.2

### Software versions
- CMake version: cmake/3.16.1
- Compilers (Intel or GNU, and version): gcc/9.3.0 gfortran
- NetCDF version:netcdf-fortran/4.5.2

**Comment by yantosca** [comment]
Hi @drewpendergrass, I think the issue might be in routine `AlarmIncrementMonths` in `History/histcontainer_mod.F90`.  It may be that there is a flaw in the logic.  I only ever tested this with regular intervals so it is possible that edge cases like 1 month and 1 day are falling through the cracks.  I'll take a look. 
```F90
!EOC
!------------------------------------------------------------------------------
!                  GEOS-Chem Global Chemical Transport Model                  !
!------------------------------------------------------------------------------
!BOP
!
! !IROUTINE: AlarmIncremenmtMonths
!
! !DESCRIPTION: Given an interval, computes the number of seconds to add
!  to an alarm, properly accounting for leap years.  This is for the case
!  when the update frequency is between 1 month and 1 year.
!\\
!\\
! !INTERFACE:
!
  SUBROUTINE AlarmIncrementMonths( IntervalYmd, Year, Month, Increment )
!
! !USES:
!
    USE History_Util_Mod, ONLY : SECONDS_PER_DAY
    USE Time_Mod,         ONLY : Its_A_Leapyear
!
! !INPUT PARAMETERS:
!
    INTEGER,  INTENT(IN)  :: IntervalYmd  ! Update frequency in YYYYMMDD format
    INTEGER,  INTENT(IN)  :: Year         ! Current year
    INTEGER,  INTENT(IN)  :: Month        ! Current month
!
! !OUTPUT PARAMETERS:
!
    REAL(f8), INTENT(OUT) :: Increment    ! Number of seconds to add to alarm
!
! !REVISION HISTORY:
!  26 Feb 2019 - R. Yantosca - Initial version
!  See https://github.com/geoschem/geos-chem for complete history
!EOP
!------------------------------------------------------------------------------
!BOC
!
! !LOCAL VARIABLES:
!
    INTEGER :: MM, nDays, nMonths, T, YYYY
!
! !DEFINED PARAMETERS:
!
    ! Days in non-leap-year months:     J  F  M  A  M  J  J  A  S  O  N  D
    INTEGER, PARAMETER :: DpM(12) = (/ 31,28,31,30,31,30,31,31,30,31,30,31 /)

    !=======================================================================
    ! AlarmIncrememntMonths begins here!
    !=======================================================================

    ! Initialize
    Increment = 0.0_fp
    MM        = Month
    nDays     = 0
    nMonths   = IntervalYmd / 100
    YYYY      = Year

    ! Loop over the requested # of months
    DO T = 0, nMonths-1

       ! Keep a running total of the number of days in the interval
       nDays = nDays + DpM(MM)

       ! Add the leap year day if necessary
       IF ( Its_A_LeapYear( YYYY ) .and. MM == 2 ) THEN
          nDays = nDays + 1
       ENDIF

       ! Increment the month for next iteration
       MM = MM + 1

       ! Also increment the year if we straddle New Year's Day
       IF ( MM > 12 ) THEN
          MM   = 1
          YYYY = YYYY + 1
       ENDIF

    ENDDO

    ! Convert from days to seconds
    Increment = DBLE( nDays ) * SECONDS_PER_DAY

  END SUBROUTINE AlarmIncrementMonths
```

**Comment by yantosca** [comment]
@drewpendergrass, this should now be fixed in commit d258b5abbcc75910b78d454aaf02dd300d3ccb7e.  I will bring this into 13.4.0 pending integration testing.

**Comment by yantosca** [comment]
We can close this issue because PR #1090 has been merged in to the 13.4.0 development stream, and has proven to be a zero-diff update.
